### PR TITLE
Creating a CWL example app

### PR DESCRIPTION
**Pull request by adthrasher**
Example app running CWL workflow on a single node. Chains BWA and Samtools to produce a mapped BAM, BAM index, and flagstat.

### CODE REVIEW

File: examples/dx-cwl-alignment/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#FROM rust:1.36.0 as fqlib-builder
+#RUN cargo install \
+#      --git https://github.com/stjude/fqlib.git \
+#      --tag v0.3.1 \
+#      --root /opt/fqlib/
+
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/Dockerfile]:
```suggestion
```

File: examples/dx-cwl-alignment/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+FROM ubuntu:18.04 as builder
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/Dockerfile]:
```suggestion
FROM ubuntu:18.04
```

File: examples/dx-cwl-alignment/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+ENV PATH /opt/conda/bin:$PATH
+
+RUN apt-get update && \
+    apt-get upgrade -y && \
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/Dockerfile]:
```suggestion
```

File: examples/dx-cwl-alignment/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    coreutils==8.31 \
+    picard==2.20.2 \
+    samtools==1.9 \
+    bwa==0.7.17 \
+    -y && \
+    conda clean --all -y
+
+RUN pip install --no-cache-dir ngsderive
+
+#COPY --from=fqlib-builder /opt/fqlib/bin/fq /usr/local/bin/
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/Dockerfile]:
```suggestion
```

File: examples/dx-cwl-alignment/build.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  echo "Version from dxapp.json is $dxapp_version, not $VERS" >&2
+  exit 1
+fi
+
+set -ex
+
+if [ "$no_docker_build" == "" ]
+then 
+  docker build -t bio-base:latest .
+  docker build -t bio-base:$VERS .
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/build.sh]:
```suggestion
  docker build -t bio-base .
  docker tag bio-base bio-base:$VERS
```

File: examples/dx-cwl-alignment/dxapp.json [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      "help": ""
+    }
+  ],
+  "outputSpec": [
+    {
+      "name": "bam",
+      "label": "Aligned BAM file",
+      "class": "file",
+      "patterns": [
+        "*"
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/dxapp.json]:
```suggestion
      "patterns": [
        "*.bam"
```

File: examples/dx-cwl-alignment/dxapp.json [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      ],
+      "help": ""
+    },
+    {
+      "name": "bam_index",
+      "label": "BAI file",
+      "class": "file",
+      "patterns": [
+        "*"
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/dxapp.json]:
```suggestion
        "*.bam.bai"
```

File: examples/dx-cwl-alignment/dxapp.json [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      ],
+      "help": ""
+    },
+    {
+      "name": "flagstat",
+      "label": "Samtools flagstat",
+      "class": "file",
+      "patterns": [
+        "*"
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/dxapp.json]:
```suggestion
        "*.flagstat.txt"
```

File: examples/dx-cwl-alignment/dxapp.json [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    "version": "1",
+    "distribution": "Ubuntu",
+    "file": "src/dx-cwl-alignment.sh",
+    "execDepends": [
+      {"name": "cwlref-runner",
+       "package_manager": "pip3"}
+    ]
+  },
+  "access": {
+    "network": ["*"],
```
Comment (zaeleus) [inline examples/dx-cwl-alignment/dxapp.json]: Does this app require full network access?

**Review by zaeleus — COMMENTED** [review]
Consider removing the Seven Bridges coordinates from the cwl definitions, as it's noisy for an example.

File: examples/dx-cwl-alignment/build.sh (hunk 5 above)
Comment (zaeleus) [reply examples/dx-cwl-alignment/build.sh]: Sorry, there shouldn't be a trailing `.` after the tag command.

File: examples/dx-cwl-alignment/dxapp.json (hunk 9 above)
Comment (adthrasher) [reply examples/dx-cwl-alignment/dxapp.json]: Probably not full, but it requires PyPi and Docker access.

**Review by jsunny23 — APPROVED** [review]

**Pull request merged**